src/registrar/assets/js/get-gov.js

```diff
@@ -994,194 +1002,382 @@ function unloadModals() {
   window.modal.off();
 }
 
-/**
+class LoadTableBase {
+  constructor(tableSelector, tableWrapperSelector, searchFieldId, searchSubmitId, resetSearchBtn, resetFiltersBtn, noDataDisplay, noSearchresultsDisplay) {
+    this.tableWrapper = document.querySelector(tableWrapperSelector);
+    this.tableHeaders = document.querySelectorAll(`${tableSelector} th[data-sortable]`);
+    this.currentSortBy = 'id';
+    this.currentOrder = 'asc';
+    this.currentStatus = [];
+    this.currentSearchTerm = '';
+    this.scrollToTable = false;
+    this.searchInput = document.querySelector(searchFieldId);
+    this.searchSubmit = document.querySelector(searchSubmitId);
+    this.tableAnnouncementRegion = document.querySelector(`${tableWrapperSelector} .usa-table__announcement-region`);
+    this.resetSearchButton = document.querySelector(resetSearchBtn);
+    this.resetFiltersButton = document.querySelector(resetFiltersBtn);
+    // NOTE: these 3 can't be used if filters are active on a page with more than 1 table
```

Comment:
They can't be used bc they are not specific enough, so interacting with them on one table will have side effects on the other table. HOWEVER, this is not an issue yet because multiple tables with filters do not co-exist. The non-org homepage has 2 tables with no filters.